src/mpid/ch4/netmod/ucx/ucx_am.h

```diff
@@ -187,19 +212,37 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_am_isendv(int rank,
     }
 
     if (dt_contig) {
-        MPIR_Memcpy(send_buf + am_hdr_sz + sizeof(ucx_hdr), (char *) data + dt_true_lb, data_sz);
+        if (data_sz < MPIR_CVAR_CH4_UCX_AM_IOV_THRESH) {
+            MPIR_Memcpy(send_buf + am_hdr_sz + sizeof(ucx_hdr), (char *) data + dt_true_lb, data_sz);
+
+            dt = ucp_dt_make_contig(1);
+            tot_sz = data_sz + am_hdr_sz + sizeof(ucx_hdr);
+        } else {
+            iov[0].buffer = send_buf;
+            iov[0].length = sizeof(ucx_hdr) + am_hdr_sz;
+            iov[1].buffer = (void *) ((char *) data + dt_true_lb);
+            iov[1].length = data_sz;
+
+            dt = ucp_dt_make_iov();
+            tot_sz = 2;
+        }
```

Comment:
this actually slows down operations such as `MPI_Accumulate`:
MPICH Master
```
# OSU MPI_Accumulate latency Test v5.6.2
# Window creation: MPI_Win_allocate
# Synchronization: MPI_Win_flush
# Size          Latency (us)
1                       2.90
2                       2.71
4                       2.68
8                       2.66
16                      2.67
32                      3.09
64                      3.06
128                     3.14
256                     3.24
512                     3.34
1024                    3.81
2048                    4.18
4096                    5.02
8192                    6.63
16384                   9.06
32768                  15.64
65536                  24.08
131072                 50.04
262144                 95.34
524288                184.32
1048576               841.29
2097152              1638.00
4194304              2837.62
```
This PR
```
# OSU MPI_Accumulate latency Test v5.6.2
# Window creation: MPI_Win_allocate
# Synchronization: MPI_Win_flush
# Size          Latency (us)
1                       2.91
2                       2.70
4                       2.68
8                       2.67
16                      2.68
32                      3.08
64                      3.12
128                     3.20
256                     3.34
512                     3.61
1024                    4.33
2048                    5.32
4096                    7.02
8192                   10.83
16384                  17.13
32768                  32.51
65536                  56.51
131072                111.61
262144                215.22
524288                422.43
1048576              1421.00
2097152              2797.93
4194304              5030.06
```